core/src/consensus/tendermint/worker.rs

```diff
@@ -996,15 +1005,20 @@ impl Worker {
             let current_step = self.step.clone();
             match current_step {
                 TendermintState::Propose => {
-                    self.move_to_step(TendermintState::Prevote, false);
+                    //self.move_to_step(TendermintState::Prevote, false);
                 }
                 TendermintState::ProposeWaitImported {
                     block,
                 } => {
                     if !block.transactions().is_empty() {
                         cinfo!(ENGINE, "Submitting proposal block {}", block.header().hash());
-                        self.move_to_step(TendermintState::Prevote, false);
-                        self.broadcast_proposal_block(self.view, encoded::Block::new(block.rlp_bytes()));
+                        self.broadcast_proposal_block(
```

Comment:
We don't need to propagate the generated block if we know another block that has a higher score.
Let's add some `FIXME` comment about it or change the behavior.

Reply:
Done